Messaging protocol

OmarQurashi868 opened the pull request:
Wrote the technical specifications for the proposed messaging protocol to be used on top of web sockets.
All details are written down in the markdown file added in this PR.

[Trello ticket](https://trello.com/c/1urQb2LJ/33-messaging-protocol)

OmarQurashi868 committed:
Initial specs

OmarQurashi868 committed:
Renamed file to LSMP

OmarQurashi868 committed:
Renamed again xd

ahmed-okkio commented:
I noticed you're using new lines, would be good to document that

ahmed-okkio commented on MessagingProtocol.md:
````diff
@@ -0,0 +1,100 @@
+how you been?
+```
+> Request to send a message from user with id `74316` in channel with id `47532`, with the message content being:  
+> `yoo good morning`  
````
I'm guessing new lines that appear after the WITH are counted within the string?

OmarQurashi868 replied:
Yes, everything after the `WITH` counts as part of the message, until EOF.

OmarQurashi868 committed:
Added details about new lines

OmarQurashi868 commented:
> I noticed you're using new lines, would be good to document that

Updated to include relevant details 👍

OmarQurashi868 committed:
Added server section, removed FROM from client requests, removed need for auth

OmarQurashi868 commented:
This is an update that implements some changes that came in accordance to new information.
## Highlights
- Clients now just need to pass in the username in the `AUTH` request instead of a token
- Removed the `FROM` parameters from requests from the client, this is because the server should save the username for the current websocket connection
- Added a CONNECT verb
- Added a section for server to client responses

OmarQurashi868 committed:
Removed old user-id references, removed channel-id usage

OmarQurashi868 commented:
- Removed old references for user-id left by accident
- Switched to using channel names instead of channel IDs

OmarQurashi868 committed:
Added the AT parameter for timestamps

OmarQurashi868 committed:
Added REMEMBER verb for message history request

OmarQurashi868 committed:
Added POPULATE server verb

OmarQurashi868 committed:
Updated version string

OmarQurashi868 commented:
Updated protocol to accommodate for message timestamps and message history requests

OmarQurashi868 committed:
Changed array seperator to `/*$*/` because $$$ is too common

ahmed-okkio commented on MessagingProtocol.md:
```diff
@@ -146,35 +146,35 @@ When the server receives a `REMEMBER` request from a client, the server is expec
    > Required params; `WITH <array-of-messages>`
 
-In the `WITH` argument, seperate message requests should start, end, and be seperated using a sequence of 3 dollar signs `$$$` like so:
+In the `WITH` argument, seperate message requests should start, end, and be seperated using a sequence of 3 dollar signs `/*$*/` like so:
```
Update paragraph

OmarQurashi868 replied:
Oof, good catch. Done

OmarQurashi868 committed:
Updated obsolete paragraph

ahmed-okkio commented:
Try to close this pr it's a bit too big now, if any more changes are needed make a new one
